analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/DoNotHardcodeCredentials.SecureString.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Security;
+
+namespace Tests.Diagnostics
+{
+    class Issue_7323
+    {
+        // https://github.com/SonarSource/sonar-dotnet/issues/7323
+        public void Noncompliant()
+        {
+            using (SecureString securePwd = new SecureString())
+            {
+                for (int i = 0; i < "AP@ssw0rd".Length; i++)
+                {
+                    securePwd.AppendChar("AP@ssw0rd"[i]); // Noncompliant {{Please review this hard-coded password.}}
+                }
+
+                // Do something with securePwd
+            }
+        }
+
+        public void Compliant(String password)
+        {
+            using (SecureString securePwd = new SecureString())
+            {
+                for (int i = 0; i < password.Length; i++)
+                {
+                    securePwd.AppendChar(password[i]); // Compliant
+                }
+
+                // Do something with securePwd
+            }
+        }
+    }
+
+    class Tests
+    {
+        public void Characters()
+        {
+            using (var securePwd = new SecureString())
+            {
+                securePwd.AppendChar('P'); // Noncompliant
+                securePwd.AppendChar('a'); // Noncompliant
+                securePwd.AppendChar('s'); // Noncompliant
+                securePwd.AppendChar('s'); // Noncompliant
```

Comment:
```suggestion
                securePwd.AppendChar('P'); // Noncompliant
                securePwd.AppendChar('a'); // Noncompliant
                securePwd.AppendChar('s'); // Noncompliant
                securePwd.AppendChar('s'); // Noncompliant
                var w = 'w';
                securePwd.AppendChar(w);   // FN
```
Adding this test will increase code coverage. Also, it needs to be fixed.

Reply:
I'm not going to fix it. We already over-delivered and we said we don't want to invest in this deprecated API.